app/com/arpnetworking/database/h2/triggers/BaseUpdateEtagTrigger.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.database.h2.triggers;
+
+import org.h2.api.Trigger;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * Base class for H2 triggers for updating etags.
+ *
+ * @author Deepika Misra (deepika at groupon dot com)
+ */
+public abstract class BaseUpdateEtagTrigger implements Trigger {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void init(
+            final Connection conn,
+            final String schemaName,
+            final String triggerName,
+            final String tableName,
+            final boolean before,
+            final int type) throws SQLException {}
+
+    /**
+     * Base method to be called by the implementing trigger's fire method.
+     *
+     * @param conn A connection to the database.
+     * @param oldRow the old row, or null if no old row is available (for INSERT)
+     * @param newRow the new row, or null if no new row is available (for DELETE)
+     * @param seqName The name of the sequence that is to be updated.
+     * @throws SQLException if the operation must be undone
+     */
+    public void baseFire(
+            final Connection conn,
+            final Object[] oldRow,
+            final Object[] newRow,
+            final String seqName) throws SQLException {
+        try (PreparedStatement stmt = conn.prepareStatement(String.format("SELECT NEXTVAL('%s');", seqName))) {
```

Comment:
By passing the argument you can format the statement on instantiation saving a string format call per invocation.